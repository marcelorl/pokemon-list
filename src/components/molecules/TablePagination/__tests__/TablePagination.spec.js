import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../TablePagination.fixture'

const { mount, getWrapper, get } = createTestContext({ fixture })

beforeEach(mount)

describe('#TablePagination', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('should run prev()', () => {
    getWrapper().find('PaginationLink').first().simulate('click')

    expect(get('props').prev).toBeCalled()
  })

  it('should run next()', () => {
    getWrapper().find('PaginationLink').last().simulate('click')

    expect(get('props').next).toBeCalled()
  })
})
