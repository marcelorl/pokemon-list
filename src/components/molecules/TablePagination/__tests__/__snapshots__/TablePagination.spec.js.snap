// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TablePagination renders component 1`] = `
<TablePagination
  next={[Function]}
  prev={[Function]}
>
  <Pagination
    size="lg"
    tag="ul"
  >
    <ul
      className="pagination pagination-lg"
    >
      <PaginationItem
        tag="li"
      >
        <li
          className="page-item"
        >
          <PaginationLink
            onClick={[Function]}
            previous={true}
            tag="a"
          >
            <a
              aria-label="Previous"
              className="page-link"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                key="caret"
              >
                «
              </span>
              <span
                className="sr-only"
                key="sr"
              >
                Previous
              </span>
            </a>
          </PaginationLink>
        </li>
      </PaginationItem>
      <PaginationItem
        tag="li"
      >
        <li
          className="page-item"
        >
          <PaginationLink
            next={true}
            onClick={[Function]}
            tag="a"
          >
            <a
              aria-label="Next"
              className="page-link"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                key="caret"
              >
                »
              </span>
              <span
                className="sr-only"
                key="sr"
              >
                Next
              </span>
            </a>
          </PaginationLink>
        </li>
      </PaginationItem>
    </ul>
  </Pagination>
</TablePagination>
`;
